private_sharing/forms.py

```diff
@@ -393,3 +420,35 @@ class DeleteDataFileForm(forms.Form):
     file_basename = forms.CharField(required=False, label="File basename")
 
     all_files = forms.BooleanField(required=False, label="All files")
+
+
+class SelectDatatypesForm(forms.Form):
+    """
+    A form that generates a tree for selecting datatypes.
+    """
+
+    def clean(self):
+        """
+        Check that something was actually passed, and, if so, populate and return
+        cleaned_data.
+        """
+        super().clean()
+        # Check to see if anything was selected
+        # wants to be on two lines:
+        data = dict(self.data)
+        data.pop("csrfmiddlewaretoken")
```

Comment:
This jumps out to me as something weird/wrong to have to do. I believe normally custom clean functions use `cleaned_data` e.g. for complex validation: https://docs.djangoproject.com/en/2.1/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other